### DISCUSSION TITLE
HP Reverb G2 / MRTK 2.7 APPX built but does not work ..

### DISCUSSION THREAD

**Discussion by sebstax7** [microsoft/MixedRealityToolkit-Unity#10031] [Q&A] [answered]
Hello,

I built an APPX, no error message in Unity, but when I launch the application it does not work.
It seems to be related to Windows mixed reality portal.

Anyone experience the bug shown below?

Using unity 2020.3.
![crash](https://user-images.githubusercontent.com/83447881/123436200-b31e0380-d5ce-11eb-9835-a0f1c0387ab4.PNG)

my player log : 

NullReferenceException: Object reference not set to an instance of an object
  at Microsoft.MixedReality.Toolkit.Utilities.Solvers.SolverHandler.AttachToNewTrackedObject () [0x0001d] in <bea29dd2c37f41089b12c59616ed020d>:0 
  at Microsoft.MixedReality.Toolkit.Utilities.Solvers.SolverHandler.RefreshTrackedObject () [0x00006] in <bea29dd2c37f41089b12c59616ed020d>:0 
  at Microsoft.MixedReality.Toolkit.Utilities.Solvers.SolverHandler.Update () [0x00008] in <bea29dd2c37f41089b12c59616ed020d>:0 

No main camera found. Searching for cameras in the scene.

The Mixed Reality Toolkit was unable to determine a main camera. Please tag the scene's primary camera as "MainCamera", in the hierarchy.

NullReferenceException: Object reference not set to an instance of an object
  at Microsoft.MixedReality.Toolkit.Utilities.Solvers.SolverHandler.AttachToNewTrackedObject () [0x0001d] in <bea29dd2c37f41089b12c59616ed020d>:0 
  at Microsoft.MixedReality.Toolkit.Utilities.Solvers.SolverHandler.RefreshTrackedObject () [0x00006] in <bea29dd2c37f41089b12c59616ed020d>:0 
  at Microsoft.MixedReality.Toolkit.Utilities.Solvers.SolverHandler.Update () [0x00008] in <bea29dd2c37f41089b12c59616ed020d>:0 

XRTextureManager::RequestDestroyTexture
XRTextureManager::RequestDestroyTexture
XRTextureManager::RequestDestroyTexture
[XR] [11032] [17:23:59.684][Info   ] OpenXRSession::HandleSessionStateChangedEvent: state XR_SESSION_STATE_VISIBLE->XR_SESSION_STATE_SYNCHRONIZED session=2 time=12188382235600
[XR] [11032] [17:23:59.684][Info   ] OpenXRSession::HandleSessionStateChangedEvent: state XR_SESSION_STATE_SYNCHRONIZED->XR_SESSION_STATE_STOPPING session=2 time=12188382245600
[XR] [11032] [17:23:59.684][Info   ] OpenXRSession::HandleSessionStateChangedEvent: state XR_SESSION_STATE_STOPPING->XR_SESSION_STATE_IDLE session=2 time=12188408597600
[XR] [11032] [17:23:59.684][Info   ] OpenXRSession::HandleSessionStateChangedEvent: state XR_SESSION_STATE_IDLE->XR_SESSION_STATE_EXITING session=2 time=12188408607000
[XR] [11032] [17:23:59.685][Info   ] 

==== Start Unity OpenXR Diagnostic Report ====

[XR] [11032] [17:23:59.685][Info   ] ==== Report Reason: System Shutdown ====

[XR] [11032] [17:23:59.685][Info   ] ==== OpenXR Support Details ====
[XR] [11032] [17:23:59.685][Info   ] OpenXR Runtime:
[XR] [11032] [17:23:59.685][Info   ]     Microsoft, which is a Unity supported partner
[XR] [11032] [17:23:59.685][Info   ] Unity OpenXR Features:
[XR] [11032] [17:23:59.685][Info   ]     HPMixedRealityControllerProfile Standalone: Microsoft, which is a Unity supported partner
[XR] [11032] [17:23:59.685][Info   ] Unity Support:
[XR] [11032] [17:23:59.685][Info   ]     Unity supports the runtime and Unity OpenXR Features above. When requesting assistance, please copy the OpenXR section from ==== Start Unity OpenXR Diagnostic Report ==== to ==== End Unity OpenXR Diagnostic Report ==== to the bug or forum post.
[XR] [11032] [17:23:59.685][Info   ] 
[XR] [11032] [17:23:59.685][Info   ] ==== OpenXR Init Info ====
[XR] [11032] [17:23:59.685][Info   ] 
[XR] [11032] [17:23:59.686][Info   ] Stage 1 Missing Entry Points: (1)
[XR] [11032] [17:23:59.686][Info   ]   xrInitializeLoaderKHR
[XR] [11032] [17:23:59.686][Info   ] 
[XR] [11032] [17:23:59.686][Info   ] Stage 2 Missing Entry Points: (9)
[XR] [11032] [17:23:59.686][Info   ]   xrGetD3D12GraphicsRequirementsKHR
[XR] [11032] [17:23:59.686][Info   ]   xrGetVulkanInstanceExtensionsKHR
[XR] [11032] [17:23:59.686][Info   ]   xrGetVulkanDeviceExtensionsKHR
[XR] [11032] [17:23:59.686][Info   ]   xrGetVulkanGraphicsDeviceKHR
[XR] [11032] [17:23:59.686][Info   ]   xrGetVulkanGraphicsRequirementsKHR
[XR] [11032] [17:23:59.686][Info   ]   xrCreateVulkanInstanceKHR
[XR] [11032] [17:23:59.686][Info   ]   xrCreateVulkanDeviceKHR
[XR] [11032] [17:23:59.686][Info   ]   xrGetVulkanGraphicsDevice2KHR
[XR] [11032] [17:23:59.686][Info   ]   xrGetVulkanGraphicsRequirements2KHR
[XR] [11032] [17:23:59.686][Info   ] 
[XR] [11032] [17:23:59.686][Info   ] ==== OpenXR Provider Info ====
[XR] [11032] [17:23:59.686][Info   ] Spec Version: 1.0.12
[XR] [11032] [17:23:59.686][Info   ] Provider Version: 1.2.2
[XR] [11032] [17:23:59.686][Info   ] App: Test MRTK 2.7 HP G2 Unity 2020.3.11 0.1 #3159629941
[XR] [11032] [17:23:59.686][Info   ] Engine: 2020.3.11f1 #3355275404
[XR] [11032] [17:23:59.686][Info   ] 
[XR] [11032] [17:23:59.686][Info   ] ==== OpenXR Runtime Info ====
[XR] [11032] [17:23:59.686][Info   ] 
[XR] [11032] [17:23:59.686][Info   ] Features requested to be enabled: 1
[XR] [11032] [17:23:59.686][Info   ]     Name=HP Reverb G2 Controller Profile Extension=XR_EXT_hp_mixed_reality_controller Version=0.9.5
[XR] [11032] [17:23:59.686][Info   ] 
[XR] [11032] [17:23:59.686][Info   ] Features with extensions that failed to be enabled: 0
[XR] [11032] [17:23:59.686][Info   ] 
[XR] [11032] [17:23:59.686][Info   ] Available Extensions: (21)
[XR] [11032] [17:23:59.686][Info   ]   Name=XR_KHR_win32_convert_performance_counter_time SpecVersion=1
[XR] [11032] [17:23:59.686][Info   ]   Name=XR_KHR_D3D11_enable SpecVersion=5
[XR] [11032] [17:23:59.686][Info   ]   Name=XR_KHR_D3D12_enable SpecVersion=7
[XR] [11032] [17:23:59.686][Info   ]   Name=XR_KHR_composition_layer_depth SpecVersion=5
[XR] [11032] [17:23:59.686][Info   ]   Name=XR_KHR_composition_layer_color_scale_bias SpecVersion=5
[XR] [11032] [17:23:59.686][Info   ]   Name=XR_MSFT_unbounded_reference_space SpecVersion=1
[XR] [11032] [17:23:59.686][Info   ]   Name=XR_MSFT_spatial_anchor SpecVersion=1
[XR] [11032] [17:23:59.686][Info   ]   Name=XR_MSFT_perception_anchor_interop SpecVersion=1
[XR] [11032] [17:23:59.686][Info   ]   Name=XR_EXT_win32_appcontainer_compatible SpecVersion=1
[XR] [11032] [17:23:59.686][Info   ]   Name=XR_EXT_conformance_automation SpecVersion=3
[XR] [11032] [17:23:59.686][Info   ]   Name=XR_MSFT_composition_layer_reprojection SpecVersion=1
[XR] [11032] [17:23:59.686][Info   ]   Name=XR_MSFT_spatial_graph_bridge SpecVersion=1
[XR] [11032] [17:23:59.686][Info   ]   Name=XR_EXT_eye_gaze_interaction SpecVersion=1
[XR] [11032] [17:23:59.686][Info   ]   Name=XR_MSFT_hand_interaction SpecVersion=1
[XR] [11032] [17:23:59.686][Info   ]   Name=XR_EXT_hand_tracking SpecVersion=4
[XR] [11032] [17:23:59.686][Info   ]   Name=XR_MSFT_hand_tracking_mesh SpecVersion=3
[XR] [11032] [17:23:59.686][Info   ]   Name=XR_MSFT_controller_model SpecVersion=2
[XR] [11032] [17:23:59.686][Info   ]   Name=XR_KHR_visibility_mask SpecVersion=2
[XR] [11032] [17:23:59.686][Info   ]   Name=XR_EXT_samsung_odyssey_controller SpecVersion=1
[XR] [11032] [17:23:59.686][Info   ]   Name=XR_EXT_hp_mixed_reality_controller SpecVersion=1
[XR] [11032] [17:23:59.686][Info   ]   Name=XR_EXT_debug_utils SpecVersion=3
[XR] [11032] [17:23:59.686][Info   ] 
[XR] [11032] [17:23:59.686][Info   ] Available Layers: (0)
[XR] [11032] [17:23:59.686][Info   ] 
[XR] [11032] [17:23:59.686][Info   ] Runtime Name: Windows Mixed Reality Runtime
[XR] [11032] [17:23:59.686][Info   ] Runtime Version: 106.2105.21002
[XR] [11032] [17:23:59.686][Info   ] 
[XR] [11032] [17:23:59.686][Info   ] Form factor: XR_FORM_FACTOR_HEAD_MOUNTED_DISPLAY
[XR] [11032] [17:23:59.686][Info   ] 
[XR] [11032] [17:23:59.686][Info   ] Spec extensions enabled: (5)
[XR] [11032] [17:23:59.686][Info   ]   Name=XR_EXT_hp_mixed_reality_controller Version=1
[XR] [11032] [17:23:59.686][Info   ]   Name=XR_KHR_D3D11_enable Version=5
[XR] [11032] [17:23:59.686][Info   ]   Name=XR_KHR_composition_layer_depth Version=5
[XR] [11032] [17:23:59.686][Info   ]   Name=XR_KHR_visibility_mask Version=2
[XR] [11032] [17:23:59.686][Info   ]   Name=XR_KHR_win32_convert_performance_counter_time Version=1
[XR] [11032] [17:23:59.686][Info   ] 
[XR] [11032] [17:23:59.686][Info   ] Available Reference Spaces: (3)
[XR] [11032] [17:23:59.686][Info   ]   Name: XR_REFERENCE_SPACE_TYPE_VIEW
[XR] [11032] [17:23:59.686][Info   ]   Name: XR_REFERENCE_SPACE_TYPE_LOCAL
[XR] [11032] [17:23:59.686][Info   ]   Name: XR_REFERENCE_SPACE_TYPE_STAGE
[XR] [11032] [17:23:59.686][Info   ] 
[XR] [11032] [17:23:59.686][Info   ] Instance Id  : 0x1
[XR] [11032] [17:23:59.686][Info   ] System Id    : 0x1
[XR] [11032] [17:23:59.686][Info   ] Session Id   : 0x2
[XR] [11032] [17:23:59.686][Info   ] 
[XR] [11032] [17:23:59.686][Info   ] Available View Configuration Types: (1)
[XR] [11032] [17:23:59.686][Info   ]   View Configuration Type: XR_VIEW_CONFIGURATION_TYPE_PRIMARY_STEREO (Selected)  View configuration FovMutable=True
[XR] [11032] [17:23:59.686][Info   ]   View [0]: Recommended Width=2488 Height=2428 SampleCount=1
[XR] [11032] [17:23:59.686][Info   ]   View [0]:     Maximum Width=6348 Height=6200 SampleCount=4
[XR] [11032] [17:23:59.686][Info   ]   View [1]: Recommended Width=2488 Height=2428 SampleCount=1
[XR] [11032] [17:23:59.686][Info   ]   View [1]:     Maximum Width=6348 Height=6200 SampleCount=4
[XR] [11032] [17:23:59.686][Info   ] 
[XR] [11032] [17:23:59.686][Info   ] Available Environment Blend Mode: (1)
[XR] [11032] [17:23:59.686][Info   ]   Environment Blend Mode (XR_ENVIRONMENT_BLEND_MODE_OPAQUE) : (Selected)
[XR] [11032] [17:23:59.686][Info   ] 
[XR] [11032] [17:23:59.686][Info   ] System Properties: Name=Windows Mixed Reality VendorId=1118
[XR] [11032] [17:23:59.686][Info   ] System Graphics Properties: MaxWidth=16384 MaxHeight=16384 MaxLayers=16
[XR] [11032] [17:23:59.686][Info   ] System Tracking Properties: OrientationTracking=True PositionTracking=True
[XR] [11032] [17:23:59.686][Info   ] 
[XR] [11032] [17:23:59.686][Info   ] ==== OpenXR Input ====
[XR] [11032] [17:23:59.686][Info   ] 
[XR] [11032] [17:23:59.686][Info   ] Action Set Data: 
[XR] [11032] [17:23:59.687][Info   ] Name:     hpmixedrealitycontroller                 Desired Profile: /interaction_profiles/hp/mixed_reality_controller
[XR] [11032] [17:23:59.687][Info   ] 
[XR] [11032] [17:23:59.687][Info   ] ==== OpenXR Swapchain Details ====
[XR] [11032] [17:23:59.687][Info   ] 
[XR] [11032] [17:23:59.687][Info   ] Render Mode: Single Pass Instanced
[XR] [11032] [17:23:59.687][Info   ] Depth Submission Mode: None
[XR] [11032] [17:23:59.687][Info   ] 
[XR] [11032] [17:23:59.687][Info   ] Swapchain Formats:  [c:29] 91 28 87 40 20 45 55
[XR] [11032] [17:23:59.687][Info   ] Creating swapchain for view 0 with dimensions Width=2488 Height=2428 SampleCount=1
[XR] [11032] [17:23:59.687][Info   ] 
[XR] [11032] [17:23:59.687][Info   ] ==== Last 20 non-XR_SUCCESS returns ====
[XR] [11032] [17:23:59.687][Info   ] [SUCCESS] xrGetReferenceSpaceBoundsRect: XR_SPACE_BOUNDS_UNAVAILABLE (3x)
[XR] [11032] [17:23:59.687][Info   ] [SUCCESS] xrPollEvent: XR_EVENT_UNAVAILABLE (200x)
[XR] [11032] [17:23:59.687][Info   ] 
[XR] [11032] [17:23:59.687][Info   ] ==== Last 20 Events ====
[XR] [11032] [17:23:59.687][Info   ] OpenXRSession::HandleSessionStateChangedEvent: XR_SESSION_STATE_UNKNOWN->XR_SESSION_STATE_IDLE
[XR] [11032] [17:23:59.687][Info   ] OpenXRSession::HandleSessionStateChangedEvent: XR_SESSION_STATE_IDLE->XR_SESSION_STATE_READY
[XR] [11032] [17:23:59.687][Info   ] OpenXRSession::HandleSessionStateChangedEvent: XR_SESSION_STATE_READY->XR_SESSION_STATE_SYNCHRONIZED
[XR] [11032] [17:23:59.687][Info   ] OpenXRSession::HandleSessionStateChangedEvent: XR_SESSION_STATE_SYNCHRONIZED->XR_SESSION_STATE_VISIBLE
[XR] [11032] [17:23:59.687][Info   ] OpenXRSession::HandleSessionStateChangedEvent: XR_SESSION_STATE_VISIBLE->XR_SESSION_STATE_SYNCHRONIZED
[XR] [11032] [17:23:59.687][Info   ] OpenXRSession::HandleSessionStateChangedEvent: XR_SESSION_STATE_SYNCHRONIZED->XR_SESSION_STATE_STOPPING
[XR] [11032] [17:23:59.687][Info   ] OpenXRSession::HandleSessionStateChangedEvent: XR_SESSION_STATE_STOPPING->XR_SESSION_STATE_IDLE
[XR] [11032] [17:23:59.687][Info   ] OpenXRSession::HandleSessionStateChangedEvent: XR_SESSION_STATE_IDLE->XR_SESSION_STATE_EXITING
[XR] [11032] [17:23:59.687][Info   ] 

==== End Unity OpenXR Diagnostic Report ====

Setting up 6 worker threads for Enlighten.
  Thread -> id: 44d0 -> priority: 1 
  Thread -> id: 5e44 -> priority: 1 
  Thread -> id: 5784 -> priority: 1 
  Thread -> id: 6b94 -> priority: 1 
  Thread -> id: 59bc -> priority: 1 
  Thread -> id: 3964 -> priority: 1

**Comment by polar-kev** [comment, marked as answer]
Hi @sebstax7, it looks like you asked the same question here: https://github.com/microsoft/MixedRealityToolkit-Unity/issues/10030
I'll mark this as answered since we're answering it there.
